src/labkey/query/Rows.ts

```diff
@@ -309,27 +353,33 @@ export interface SaveRowsOptions extends RequestCallbackOptions<SaveRowsResponse
  */
 export function saveRows(options: SaveRowsOptions): XMLHttpRequest {
     // Nick: I've elected to comment this out as saveRows never supported the same argument
-    // pattern as other endpoints due to the different nature of it's arguments (e.g. doesn't take a
+    // pattern as other endpoints due to the different nature of its arguments (e.g. doesn't take a
     // schema/query but rather commands, etc). As a result, the object would not match what is expected.
     // if (arguments.length > 1) {
     //     options = queryArguments(arguments);
     // }
-
-    return request({
+    const jsonData = {
+        apiVersion: options.apiVersion,
+        commands: options.commands,
+        containerPath: options.containerPath,
+        extraContext: options.extraContext,
+        transacted: options.transacted,
+        validateOnly: options.validateOnly,
+    };
+    const requestOptions: RequestOptions = {
         url: buildURL('query', 'saveRows.api', options.containerPath),
         method: 'POST',
-        jsonData: {
-            apiVersion: options.apiVersion,
-            commands: options.commands,
-            containerPath: options.containerPath,
-            extraContext: options.extraContext,
-            transacted: options.transacted,
-            validateOnly: options.validateOnly,
-        },
         success: getCallbackWrapper(getOnSuccess(options), options.scope),
         failure: getCallbackWrapper(getOnFailure(options), options.scope, true),
         timeout: options.timeout,
-    });
+    };
+
+    const form = bindSaveRowsData(jsonData);
+
+    if (form !== undefined) requestOptions.form = form;
```

Comment:
This warrants unit tests in `Rows.spec.ts`. See the tests for other endpoint wrappers in that test file for how the mocking is done. Also of interesting is the `bindFormData` test coverage which can be helpful guide in adding tests for these new behaviors.

Reply:
Done